plumbing/cst/chain_sync.go

```diff
@@ -37,3 +48,26 @@ func (chs *ChainSyncProvider) Status() chain.Status {
 func (chs *ChainSyncProvider) HandleNewTipSet(ctx context.Context, ci *types.ChainInfo, trusted bool) error {
 	return chs.sync.HandleNewTipSet(ctx, ci, trusted)
 }
+
+// ChainImport imports a chain with the same genesis block from `in`.
+func (chs *ChainSyncProvider) ChainImport(ctx context.Context, in io.Reader) error {
```

Comment:
In my view this does too much.  To keep separation of concerns and a predictable CLI I want `import` to wrap only the `chain.Import` call and return the headKey.  Then I want the `HandleNewTipSet` call to be the single target of a new `chain sync` command.

Ideally `chain export` would also take in a tipset key for exporting and users would run `chain head` first to get that key to get the behavior we have here.  This is less important to me than separating data IO from scheduling sync jobs.

Reply:
I concur with all of these.

Importing is something one does with a chain store, not a syncer. It should be parallel with export. The `chain sync` command is then already sufficient to switch to the imported head if heavier.